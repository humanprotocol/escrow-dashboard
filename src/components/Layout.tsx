import * as React from 'react';
import { Box } from '@mui/material';
import Header from './Header';

interface ILayout {
  children: React.ReactNode;
}

const Layout: React.FC<ILayout> = ({ children }): React.ReactElement => (
  <Box>
    <Header />
    {children}
  </Box>
);

export default Layout;
