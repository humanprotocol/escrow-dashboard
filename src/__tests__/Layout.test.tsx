import React from 'react';
import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';
import Layout from 'src/components/Layout';

describe('when rendered with a `name` prop', () => {
  it('should paste it into the greetings text', () => {
    const { getByRole, getByTestId } = render(
      <Layout>
        <div role="root">
          <div data-testid="parent">
            <div data-testid="child">content</div>
          </div>
        </div>
      </Layout>
    );

    const root = getByRole('root');
    const parent = getByTestId('parent');
    const child = getByTestId('child');
    expect(root).toContainElement(parent);
    expect(parent).toContainElement(child);
    expect(child).not.toContainElement(parent); // Pass
  });
});

it('renders correctly', () => {
  const tree = renderer
    .create(
      <Layout>
        <div role="root">
          <div data-testid="parent">
            <div data-testid="child">content</div>
          </div>
        </div>
      </Layout>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
