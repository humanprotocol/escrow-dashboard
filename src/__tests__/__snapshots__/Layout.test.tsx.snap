// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="MuiBox-root css-0"
>
  <div
    className="MuiBox-root css-i9gxme"
  >
    <header
      className="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionFixed mui-fixed css-11b3ww9-MuiPaper-root-MuiAppBar-root"
    >
      <div
        className="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-hyum1k-MuiToolbar-root"
      >
        <div
          className="MuiBox-root css-1o0y1dn"
        >
          <div
            className="MuiTypography-root MuiTypography-h6 css-vr6w4y-MuiTypography-root"
          >
            HUMAN Escrow Factory Dashboard
          </div>
        </div>
      </div>
    </header>
  </div>
  <div
    role="root"
  >
    <div
      data-testid="parent"
    >
      <div
        data-testid="child"
      >
        content
      </div>
    </div>
  </div>
</div>
`;
